src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -196,3 +192,32 @@ private func operationName(for operationName: String) -> String {
     }
     return nameComps.joined()
 }
+
+/// Build a list of required and optional query params and headers from a list of parameters
+private func buildQueryParamsHeaders(
+    parameters: [Parameter]?,
+    with operation: Operation,
+    requiredQueryParams: inout [KeyValueViewModel],
+    requiredHeaders: inout [KeyValueViewModel],
+    optionalQueryParams: inout [KeyValueViewModel],
+    optionalHeaders: inout [KeyValueViewModel],
+    pathParams: inout [KeyValueViewModel]
```

Comment:
Do these not have initial values?  If all we are doing is processing parameters into a structure that is more manageable, then I would recommend the approach I suggested above. Another alternative that is messier but avoids the use of `inout` (which is discouraged anyways) is to return a tuple.

Reply:
Will implement without `inout`